internal/server/webservice/webservice_test.go

```diff
@@ -415,3 +404,45 @@ func newForTest(t *testing.T, cm *testConfigManager, daemonConfig *webservice.St
 	require.NoError(t, err, "Setup: failed to create server")
 	return s
 }
+
+func newAndRun(t *testing.T, cm *testConfigManager, daemonConfig *webservice.StaticConfig) (*webservice.Server, chan error) {
+	t.Helper()
+
+	muPortAcquire.Lock()
+	defer muPortAcquire.Unlock()
+
+	s := newForTest(t, cm, daemonConfig)
+	t.Cleanup(func() {
+		s.Quit(true)
+	})
+
+	runErr := make(chan error, 1)
+	go func() {
+		defer close(runErr)
+		runErr <- s.Run()
```

Comment:
It'd be nice to claim the port before `s.Run()` but I'm not even sure that's possible.